xun/functions/driver/dask.py

```diff
@@ -1,52 +1,127 @@
+from .. import graph as graph_helpers
 from .driver import Driver
-import dask
+import asyncio
+import functools
 import logging
 import networkx as nx
 
 logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
 
 
 class DaskDriverError(Exception):
     pass
 
 
-def compute_proxy(node, dependencies, func, store_accessor):
-    if store_accessor.completed(node):
-        return store_accessor.load_result(node)
+class Dask(Driver):
+    def __init__(self, client):
+        self.client = client
+
+    def _exec(self, graph, entry_call, function_images, store_accessor):
+        assert nx.is_directed_acyclic_graph(graph)
+        scheduler = DaskSchedule(self.client, function_images, store_accessor)
+        return scheduler(entry_call, graph)
 
+
+def compute_proxy(store_accessor, func, node):
     args, kwargs = store_accessor.resolve_call_args(node)
-    try:
-        result = func(*args, **kwargs)
-    except Exception as e:
-        raise DaskDriverError(f'task {node} failed') from e
-    except:
-        raise ValueError('???')
```

Comment:
Why was try/except mechanism removed from here? It could have been used for the case if storing is not succesfull even though the execution goes smoothly.